src/scripts/modules/media/media.coffee

```diff
@@ -94,10 +94,23 @@ define (require) ->
       mainPage.regions.main.append(footerNav)
       footerNav.$el.addClass('footer-nav')
       @mainContent = windowWithSidebar.regions.main
-
       $pinnable = @regions.pinnable.$el
       pinnableTop = $pinnable.offset().top
       $toc = tocView.$el
+      
+      tocModel = tocView.model
+      @listenTo(tocModel, 'add:contents change:contents', ->
+        mediaBodyView.on('render', ->
+          $totalSubtitle = $("body").find("section [data-depth=1]")
+          subTable=$.map($totalSubtitle, (node) ->
+            subTitles=$(node).find('> [data-type="title"]').clone().text()
+            subIds="#" + $(node).attr("id")
+            return {text:subTitles,id:subIds}
+          )
+          tocModel.get('contents').models[1].attributes.contents.models[1].set({subTable: subTable})
```

Comment:
Actually, I am wrong with these two subscripts. These two subscripts should correspond to indexes of current model and current page. I made some changes and added some comments.
And According to Ed and Ross's requirement, I don't need to add additional TOC for page that is not in a book. After my change, there is no error, just nothing happens because such kind of page doesn't have table of contents.